arb_os/gasAccounting.mini

```diff
@@ -785,28 +787,31 @@ impure func payForTxAndL1Gas(
     );
 }
 
-impure func updateCongestionState(state: CongestionState, blockNum: uint) -> CongestionState {
-    // update the congestion state, based on advancing from state.lastBlockNumSeen to blockNum
+impure func updateCongestionState(state: CongestionState, toBlockNum: uint, toTimestamp: uint) -> CongestionState {
+    // update the congestion state, based on advancing from state.lastTimestampSeen to toTimestamp
 
-    let blocksToAdvance = blockNum - state.lastBlockNumSeen;
+    let lastTimestampSeen = state.lastTimestampSeen;
     let congestionPrice = state.congestionPriceWei;
     let basePrice = gasAccountingInfo.currentPrices.perArbGasBase;
     let gasPool = state.gasPool;
-    let speedLimit = gasAccountingParams.SpeedLimitPerBlock;
-    while((blocksToAdvance > 0) && ( (state.gasPool < gasAccountingParams.GasPoolMax) || (congestionPrice > 0) )) {
-        blocksToAdvance = blocksToAdvance-1;
+    let gasPoolMax = gasAccountingParams.GasPoolMax;
+    let speedLimit = gasAccountingParams.SpeedLimitPerSecond;
+
+    //TODO: solve this analytically, so we don't have to do a per-second loop
```

Comment:
Noting the TODO here

Reply:
I updated the comment, and tightened up the code a bit.  I think the cost of this is minimal, maybe a few hundred ArbGas per second at most.  100 ArbGas is 0.0001% of the chain's capacity.

Reply:
Haha sounds good